ethers-signers/src/lib.rs

```diff
@@ -93,6 +95,21 @@ pub trait Signer: std::fmt::Debug + Send + Sync {
     /// Signs the transaction
     async fn sign_transaction(&self, message: &TypedTransaction) -> Result<Signature, Self::Error>;
 
+    /// Encodes and signs the typed data according EIP-712
+    /// payload must implement Eip712 trait.
+    ///
+    /// NOTE: Returning is Option<Signature> instead of Signature
+    /// due to use of std::convert::Infallible as an error type
+    /// in the Wallet implementation, which conflicts with
+    /// the error types used within the method, which cannot be mapped
+    /// to Infallible type. To avoid breaking changes and changing the error
+    /// type, this method returns `None` if there is an error that restricts
+    /// the payload from being encoded, and signed.
```

Comment:
Mhm this is a bit weird, we should be able to do it normally

Reply:
Still looking for a better solution than the following:

```rust
async fn sign_typed_data<T: Eip712 + Send + Sync>(
        &self,
        payload: T,
    ) -> Result<Signature, Self::Error> {
        let encoded = match payload.encode_eip712() {
            Ok(e) => e,
            Err(_) => {
                // NOTE: Still looking for a better solution for mapping to Infallible error type;
                unreachable!()
            }
        };

        Ok(self.sign_hash(H256::from(encoded), false))
    }
```